utils/utils.go

```diff
@@ -0,0 +1,19 @@
+package utils
+
+import dagpb "github.com/ipld/go-codec-dagpb"
+
+// Lookup finds a name key in a list of dag pb links
+func Lookup(links dagpb.PBLinks, key string) dagpb.Link {
+	li := links.Iterator()
+	for !li.Done() {
+		_, next := li.Next()
+		name := ""
+		if next.FieldName().Exists() {
+			name = next.FieldName().Must().String()
+		}
+		if key == name {
```

Comment:
Shouldn't this be done inside the previous `if` block?

Reply:
this is old code, but technically the idea is you pass empty string you get the first node with no name. There are reasons to do this, namely I believe it to most closely approximate existing functionality.